### PR TITLE
Generalize benchmark code for future benchmark tests

### PR DESCRIPTION
**Pull request by annarudy**
In order to create an entire repository of bq benchmark tests, the current code needed to be modified in the following ways: 
- a new schema was created to capture a subset of fields for all benchmarks, and another subset for the type of job a given benchmark is testing (load, query, etc). 
- the BenchmarkResultsUtil was modified to become a parent class that will handle all generic fields. Then a LoadBenchmarkResultsUtil was created to handle job specific field results and inherit from BenchmarkResultsUtil. 
- all other classes were separated two directories: one for classes that could be used by any type of benchmark in the future and another for classes that are specific to load jobs.

### CODE REVIEW

File: examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_result_util.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+            "location": self.job.location,
+            "startTime": self.job.started.isoformat(),
+            "endTime": self.job.ended.isoformat(),
+            "duration": (self.job.ended - self.job.started).total_seconds(),
+            "totalSlotMs": total_slot_ms,
+            "avgSlots": avg_slots
+        }
+
+    def insert_results_row(self):
+        """Gathers the results of a load job into a benchmark table.
```
[the remark this answers is not in the document]
Comment (annarudy) [reply examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_result_util.py]: good point, yes!

File: examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_parameters.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"""benchmark_parameters.py holds a dict called BENCHMARK_PARAMETERS that holds
```
[the remark this answers is not in the document]
Comment (annarudy) [reply examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_parameters.py]: Yeah I wasn't totally sure on this one, but the idea was to keep track of all the benchmarks taking place and which types of jobs they're performing in a universal dict, and to make sure the job types are listed as either 'LOAD', 'QUERY', 'COPY', or 'EXTRACT'. So far we have the file loader benchmark, which is a 'LOAD' job. In [line 197 of benchmark_load_table.py](https://github.com/GoogleCloudPlatform/professional-services/pull/355/files#diff-97a14cf6557721147065fbdca68f861aR197), I refer to the dict to get the job_type from the universal dict. I could just say `job_type = 'LOAD' ` in that line, though.

File: examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_result_util.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+from google.cloud import bigquery
+from google.cloud import storage
+
+from generic_benchmark_tools import file_constants
+from generic_benchmark_tools import table_util
+
+BYTES_IN_MB = 1000000
+
+
+class BenchmarkResultUtil(object):
```
[the remark this answers is not in the document]
Comment (annarudy) [reply examples/bq_file_load_benchmark/generic_benchmark_tools/benchmark_result_util.py]: I agree on this. The actually running of the benchmark takes place outside of this class, so i didn't include an abstract `run_benchmark`. Instead, I added an abstract `_set_job_properties` where job specific results can be set in addition to the instantiated `_set_generic_properties` in the parent class.

**Comment by annarudy** [comment]
Hey @jaketf, looking back at this, it looks like you were good with all the changes right? Can we merge and close this out?

**Comment by jaketf** [comment]
yup!

**Pull request merged**